src/main/java/com/google/sps/data/AvailabilityTimeSlotGenerator.java

```diff
@@ -0,0 +1,158 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.TextStyle;
+import java.util.Locale;
+import java.time.format.DateTimeFormatter;
+
+// TODO: Add a test for the constructor.
+
+/** A generator of a collection of AvailabilityTimeSlot Objects. */
+public class AvailabilityTimeSlotGenerator {
+  private static final int numberOfSlotsPerDay = 48;
```

Comment:
Why is this decoupled from `generateTimes()` which actually determines how many slots there are?

(This probably should not be a constant, but rather just size() of generateTimes(), which should probably only be calculated once.)

Reply:
Done